libnavigation-ui/src/main/java/com/mapbox/navigation/ui/instruction/GuidanceViewImageProvider.kt

```diff
@@ -83,17 +66,20 @@ class GuidanceViewImageProvider() {
 
     /**
      * The API allows you to cancel the rendering of guidance view.
-     * @param context Context
      */
-    fun cancelRender(context: Context) {
-        if (targets.isNotEmpty()) {
-            for(target in targets) {
-                Picasso.Builder(context).build().cancelRequest(target)
-            }
-            targets.clear()
+    fun cancelRender() {
+        if (coroutineScope.isActive) {
+            coroutineScope.cancel()
         }
     }
 
+    private fun getBitmapAsync(url: String): Deferred<Bitmap> =
+        coroutineScope.async(Dispatchers.IO) {
+            val req = Request.Builder().url(url).build()
+            val res = okHttpClient.newCall(req).execute()
+            BitmapFactory.decodeStream(res.body()?.byteStream())
+        }
```

Comment:
there is no sense to use `async{...}` and immediately call `await`. 
if you need to run some operations in parallel, you should have
```
val a = async{...}
val b = async{...}
println(a.await, b.await)
```


`getBitmapAsync ` is just switches a context.
so, you should use 
```
private fun getBitmap(url: String) : Bitmap = withContext(ThreadController.IODispatcher){...}
```